### PR TITLE
MaskedTextField: fixing the controlled behavior

### LINKED ISSUE #8482: MaskedTextField doesn't update with a controlled value
### Environment Information 
 
- __Package version(s)__: 6.161.0

### Please provide a reproduction of the bug in a codepen:
 
https://codepen.io/anon/pen/LaoWzG
 
#### Actual behavior:
 
Value is not properly updated when being externally controlled.
 
#### Expected behavior:
 
Value is updated

#### Priorities and help requested:
 
Are you willing to submit a PR to fix? Yes
 
Requested priority: Blocking
 
Products/sites affected: (if applicable) MS DNA Storage
 
#### Area of issue:

https://github.com/OfficeDev/office-ui-fabric-react/blob/master/packages/office-ui-fabric-react/src/components/TextField/MaskedTextField/MaskedTextField.tsx#L92

#### Requested fix:

Currently:
```
if (newProps.mask !== this.props.mask) {
  this._maskCharData = parseMask(newProps.mask, newProps.maskFormat);
  this.state = {
    displayValue: getMaskDisplay(newProps.mask, this._maskCharData, newProps.maskChar)
  };
}
```

Update to:
```
if (newProps.mask !== this.props.mask || newProps.value !== this.props.value) {
  this._maskCharData = parseMask(newProps.mask, newProps.maskFormat);

  newProps.value && this.setValue(newProps.value);

  this.setState({
    displayValue: getMaskDisplay(newProps.mask, this._maskCharData, newProps.maskChar)
  });
}
```

### PR DESCRIPTION
**Pull request by Vitalius1**
#### Pull request checklist

- [X] Addresses an existing issue: Fixes #8482 
- [X] Include a change request file using `$ npm run change`

#### Description of changes

When the `MaskedTextField` is used as a controlled input it was failing to update the value properly.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/8529)

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/TextField/MaskedTextField/MaskedTextField.tsx [hunk 1]
```diff
@@ -90,11 +90,13 @@ export class MaskedTextField extends BaseComponent<ITextFieldProps, IMaskedTextF
   }
 
   public componentWillReceiveProps(newProps: ITextFieldProps) {
-    if (newProps.mask !== this.props.mask) {
+    if (newProps.mask !== this.props.mask || newProps.value !== this.props.value) {
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/TextField/MaskedTextField/MaskedTextField.tsx]: Can you add a test that validates that passing in a new value correctly updates the state?

**Comment by dzearing** [comment]
Thanks for fixing this, can you add a test case to cover this behavior?

**Comment by ecraig12345** [comment]
@lambertwang should also take a look at this if possible.

File: packages/office-ui-fabric-react/src/components/TextField/MaskedTextField/MaskedTextField.tsx (hunk 1 above)
Comment (Vitalius1) [reply packages/office-ui-fabric-react/src/components/TextField/MaskedTextField/MaskedTextField.tsx]: I will. 👍

**Comment by Vitalius1** [comment]
@dzearing I added a test as requested. Can you take a look, please? Thanks!

**Comment by khmakoto** [comment]
@Vitalius1 it seems you left an unused variable in `MaskedTextField.test.tsx` that's preventing your build from succeeding.

File: packages/office-ui-fabric-react/src/components/TextField/MaskedTextField/MaskedTextField.tsx [hunk 2]
```diff
@@ -90,11 +90,13 @@ export class MaskedTextField extends BaseComponent<ITextFieldProps, IMaskedTextF
...
       this._maskCharData = parseMask(newProps.mask, newProps.maskFormat);
-      this.state = {
+      newProps.value && this.setValue(newProps.value);
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/TextField/MaskedTextField/MaskedTextField.tsx]: if newProps.value went from "sdfasfdsf" to "", this wouldn't execute.

Comment (dzearing) [reply packages/office-ui-fabric-react/src/components/TextField/MaskedTextField/MaskedTextField.tsx]: this should be comparing `newProps.value !== undefined`

Comment (ArrayKnight) [reply packages/office-ui-fabric-react/src/components/TextField/MaskedTextField/MaskedTextField.tsx]: Be sure to update the constructor with the same logic/check:
https://github.com/OfficeDev/office-ui-fabric-react/blob/be33ed02d7e712e59c43d56e7580c288aa488ffe/packages/office-ui-fabric-react/src/components/TextField/MaskedTextField/MaskedTextField.tsx#L82

**Pull request merged**